configure.ac

```diff
@@ -1605,6 +1605,16 @@ if test "${enable_chipoble_bluez}" = "yes"; then
 
     AX_CHECK_COMPILER_OPTIONS([C],   ${GLIB_CFLAGS} ${DBUS_CFLAGS})
     AX_CHECK_COMPILER_OPTIONS([C++], ${GLIB_CFLAGS} ${DBUS_CFLAGS})
+
```

Comment:
We need to move GIO-UNIX right after the GIO module check since this is not BlueZ specific

if test "${CHIP_DEVICE_LAYER_TARGET_LINUX}" = 1; then
    PKG_CHECK_MODULES([GIO], [gio-2.0])

The following parts are extra since they are defined by PKG_CHECK_MODULES, refer to GIO module check
    AC_SUBST(GIOUNIX_CFLAGS)
    AC_SUBST(GIOUNIX_LIBS)
    AX_CHECK_COMPILER_OPTIONS([C],   ${GIOUNIX_CFLAGS})
    AX_CHECK_COMPILER_OPTIONS([C++], ${GIOUNIX_CFLAGS})

Reply:
Yep, just relocated.

AC_SUBST(GIOUNIX_CFLAGS)
AC_SUBST(GIOUNIX_LIBS)
AX_CHECK_COMPILER_OPTIONS([C], ${GIOUNIX_CFLAGS})
AX_CHECK_COMPILER_OPTIONS([C++], ${GIOUNIX_CFLAGS})

they are still needed, without that, it will complain the below error
gen/DbusBluez.c:17:12: fatal error: gio/gunixfdlist.h: No such file or directory
   17 | #  include <gio/gunixfdlist.h>
      |            ^~~~~~~~~~~~~~~~~~~